packages/ripple-keypairs/src/index.ts

```diff
@@ -95,6 +95,15 @@ function deriveAddress(publicKey: string): string {
   return deriveAddressFromBytes(hexToBytes(publicKey))
 }
 
+function derivePublicKey(privateKey: string): string {
+  const algorithm = getAlgorithmFromPrivateKey(privateKey)
+
+  if (algorithm === 'ecdsa-secp256k1') {
+    return secp256k1.deriveKeypairFromPrivateKey(privateKey).publicKey
+  }
+  return ed25519.deriveKeypairFromPrivateKey(privateKey).publicKey
```

Comment:
Can you add an explicit if-condition that does:

```
if (algorithm === 'ed25519') {
    return ed25519.deriveKeypairFromPrivateKey(privateKey).publicKey
  } else
throw new Error('Unknown signing scheme algorithm')
```

In the future, if support for new SigningSchemes are provided in the xrpl.js library, then this code is protected from incorrect default behavior.

Reply:
I didn't do this, because it was causing a linting issue. 

I have added it and suppressed the warning.
https://github.com/interc0der/xrpl.js/blob/d6ba366155e56cbd9e736efcd9073717c01b7b69/packages/ripple-keypairs/src/index.ts#L98-L108